punishments.js

```diff
@@ -1114,16 +1113,30 @@ Punishments.checkIp = function (user, connection) {
 		}
 	}
 
-	Dnsbl.reverse(ip).then(host => {
+	Dnsbl.reverse(ip).catch(e => {
+		// If connection.user is reassigned before async tasks can run, user
+		// may no longer be equal to it.
+		user = connection.user || user;
+		if (e.code === 'EINVAL') {
+			if (!user.locked && !user.autoconfirmed) {
+				user.semilocked = '#dnsbl';
+			}
+		} else {
+			throw e;
+		}
+		return null;
```

Comment:
Is the return value here meaningful?

Reply:
It's meant to be used by `Chat.hostfilters` to be able to tell when connections' IPs are invalid, in case they need their own filter. Taking out the return and checking if the host is undefined instead would still work, but I think this makes it a bit clearer

Reply:
I don't know if I really like the `if-else` syntax here. I'll refactor it a bit after merging.

Reply:
ae3c9c6170c834